src/main/java/core/basesyntax/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.mapvalidator.KeyAbsentValidator;
+import java.util.Map;
+
+public class BalanceOperationHandler
+        extends AbstractOperationHandler
+        implements KeyAbsentValidator {
+    public BalanceOperationHandler() {
+        super(Operation.BALANCE);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                      FruitTransaction transaction) {
```

Comment:
Follow the encapsulation principle. The method `processTransaction` should have the lowest possible visibility that is required. As it is part of the implementation details, consider making it package-private if it's not intended to be used outside of the package.